proactive-support-bot.py

```diff
@@ -28,6 +32,16 @@ def handle_message(event_data):
     text = message.get('text')
     if not text:
         return
+
+    timestamp = float(message.get('ts', 0))
+    if timestamp in recent_events:  # high-resolution timestamps should have few false-negatives
+        logger.info('ignoring duplicate message: {}'.format(message))
+        return
+
+    recent_events.add(timestamp)  # add after check without a lock should be a small race window
```

Comment:
recent_events is undefined here. add `global recent_events` at the top

Reply:
> `recent_events` is undefined here. add `global recent_events` at the top

Done with 5d81b2a -> 808136e.